### PR TITLE
fix: Remove setTopic from GuildChannel and apply to TextChannel

### PR DESCRIPTION
**Pull request by zaida04**
#4831 
Remove setTopic from GuildChannel
Add it to TextChannel ~~(or TextBasedChannel??)~~
NewsChannel inherits it from TextChannel.

**Status**

- [ ] Code changes have been tested against the Discord API, or there are no code changes
- [X] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
- [X] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by monbrey** [comment]
If you're fixing incorrect inheritence, I'd also point out that CategoryChannel.createInvite and CategoryChannel.fetchInvites don't work, as categories cannot have invites.

File: typings/index.d.ts [hunk 1]
```diff
@@ -1527,6 +1526,7 @@ declare module 'discord.js' {
     ): Promise<Webhook>;
     public setNSFW(nsfw: boolean, reason?: string): Promise<TextChannel>;
     public setRateLimitPerUser(rateLimitPerUser: number, reason?: string): Promise<TextChannel>;
+    public setTopic(topic: string, reason?: string): Promise<TextChannel>;
```
Comment (tguichaoua) [inline typings/index.d.ts:1529]:
```suggestion
    public setTopic(topic: string, reason?: string): Promise<this>;
```
If the methods is called on `NewsChannel` you will have the correct returned type with `this`.

**Review by tguichaoua — CHANGES_REQUESTED** [review]

File: typings/index.d.ts (hunk 1 above)
Comment (zaida04) [reply typings/index.d.ts]: Fixed. Thank you!

Comment (advaith1) [reply typings/index.d.ts]: Isn't this broken on all methods then?

File: src/structures/CategoryChannel.js [hunk 2]
```diff
@@ -7,6 +7,14 @@ const GuildChannel = require('./GuildChannel');
  * @extends {GuildChannel}
  */
 class CategoryChannel extends GuildChannel {
+  _patch(data) {
+    super._patch(data);
+
+    // Category Channels cannot have invites
+    this.createInvite = undefined;
+    this.fetchInvites = undefined;
```
Comment (NotSugden) [inline src/structures/CategoryChannel.js:15]: typings need updating for this, and perhaps this should be done in the constructor, and not every time the channel is updated

would it perhaps be an idea to use an interface (similar to `TextBasedChannel`) to apply these methods in the first place? idk

Comment (zaida04) [reply src/structures/CategoryChannel.js:15]: Gotcha, I took inspiration for that from NewsChannel, which does it in the _patch method

File: src/structures/CategoryChannel.js [hunk 3]
```diff
@@ -7,6 +7,14 @@ const GuildChannel = require('./GuildChannel');
  * @extends {GuildChannel}
  */
 class CategoryChannel extends GuildChannel {
+  constructor(guild, data) {
+    super(guild, data);
+
+    // Category Channels cannot have invites
+    this.createInvite = undefined;
```
Comment (vladfrangu) [inline src/structures/CategoryChannel.js:14]: Is there even any need for this? You moved the base function from GuildChannel to TextChannel, CategoryChannel inherits from GuildChannel, not TextChannel

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: src/structures/CategoryChannel.js (hunk 3 above)
Comment (zaida04) [reply src/structures/CategoryChannel.js:14]: If you’re referring to my setting of those methods to undefined, this was a solution to another problem brought up about category channels improperly inheriting methods https://github.com/discordjs/discord.js/pull/4832#issuecomment-695605097

Comment (vladfrangu) [reply src/structures/CategoryChannel.js:14]: Oops, missed that, you're right

**Review by vladfrangu — APPROVED** [review]

**Comment by iCrawl** [comment]
This needs its conflicts resolved.

File: src/structures/CategoryChannel.js [hunk 4]
```diff
@@ -7,6 +7,14 @@ const GuildChannel = require('./GuildChannel');
  * @extends {GuildChannel}
  */
 class CategoryChannel extends GuildChannel {
+  constructor(guild, data) {
```
Comment (SpaceEEC) [inline src/structures/CategoryChannel.js:10]: I don't think these changes belong into a PR with the title `fix: Remove setTopic from GuildChannel and apply to TextChannel`.

Also, I do not think that documented methods should be `undefined`, I'd rather have these behave similar to `PartialGroupDMChannel#delete` and `#fetch`: (Make them throw an error instead)
https://github.com/discordjs/discord.js/blob/9ffcd83027f0fc06d69df21475865ad55138de01/src/structures/PartialGroupDMChannel.js#L37-L43
We could additionally to that add a warning to the docstrings of these methods in the `CategoryChannel` class.

**Review by SpaceEEC — CHANGES_REQUESTED** [review]

File: src/structures/CategoryChannel.js (hunk 4 above)
Comment (zaida04) [reply src/structures/CategoryChannel.js:10]: Wellll, this is consistent with https://github.com/discordjs/discord.js/blob/master/src/structures/NewsChannel.js#L14 where the same thing is done.

Comment (SpaceEEC) [reply src/structures/CategoryChannel.js:10]: That is a property, not a method.

**Comment by kyranet** [comment]
There are merge conflicts, please rebase or resolve them, @zaida04.

Furthermore, a solution for @SpaceEEC's requested change awaits. I think that after that has been addressed, the PR will look good to go.

**Comment by zaida04** [comment]
Closing as this PR is on a now deleted fork of djs which means I can no longer push updates to this. Will reopen on new fork.

**Pull request closed without merging**